internal/archive/archive.go

```diff
@@ -98,76 +106,83 @@ func openUbuntu(options *Options) (Archive, error) {
 	}
 
 	logf("Fetching %s %s archive details...", options.Label, options.Version)
-	reader, err := archive.fetch("Release", "")
-	if err != nil {
-		return nil, err
-	}
 
-	ctrl, err := control.ParseReader("Label", reader)
-	if err != nil {
-		return nil, fmt.Errorf("parsing archive Release file: %v", err)
-	}
-	section := ctrl.Section("Ubuntu")
-	if section == nil {
-		return nil, fmt.Errorf("corrupted archive Release file: no Ubuntu section")
-	}
-	logf("Release date: %s", section.Get("Date"))
-
-	archive.release = section
-	archive.packages = make(map[string]control.File)
+	archive.packages = make(map[string][]control.File)
 	//archive.components = strings.Fields(section.Get("Components"))
 	archive.components = []string{"main", "universe"}
-
-	digests := archive.release.Get("SHA256")
-	for _, component := range archive.components {
-		packagesPath := fmt.Sprintf("%s/binary-%s/Packages", component, options.Arch)
-		digest, _, _ := control.ParsePathInfo(digests, packagesPath)
-		logf("Fetching %s %s %s component...", options.Label, options.Version, component)
-		reader, err = archive.fetch(packagesPath+".gz", digest)
+	for _, suite := range archive.suites {
+		reader, err := archive.fetch(suite, "Release", "")
 		if err != nil {
 			return nil, err
 		}
 
-		ctrl, err := control.ParseReader("Package", reader)
+		ctrl, err := control.ParseReader("Label", reader)
 		if err != nil {
-			return nil, fmt.Errorf("parsing archive Package file: %v", err)
+			return nil, fmt.Errorf("parsing archive Release file: %v", err)
+		}
+		section := ctrl.Section("Ubuntu")
+		if section == nil {
+			return nil, fmt.Errorf("corrupted archive Release file: no Ubuntu section")
+		}
+		logf("Release date for %s: %s", suite, section.Get("Date"))
+
+		archive.release = section
+
+		digests := archive.release.Get("SHA256")
+		for _, component := range archive.components {
+			packagesPath := fmt.Sprintf("%s/binary-%s/Packages", component, options.Arch)
+			digest, _, _ := control.ParsePathInfo(digests, packagesPath)
+			logf("Fetching %s %s %s %s component...", options.Label, options.Version, suite, component)
+			reader, err = archive.fetch(suite, packagesPath+".gz", digest)
+			if err != nil {
+				return nil, err
+			}
+
+			ctrl, err := control.ParseReader("Package", reader)
+			if err != nil {
+				return nil, fmt.Errorf("parsing archive Package file: %v", err)
+			}
+			archive.packages[component] = append(archive.packages[component], ctrl)
```

Comment:
This would be solved by https://github.com/canonical/chisel/issues/20 @niemeyer What do you think about that approach? It mirrors how Apt sources are specified.